From 1a60a633365ea0981f321ca14563de017bf15dd6 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:23:24 -0400
Subject: [tools] perf/build: Add 'autodep' functionality, generate feature test dependencies automatically

Message-id: <1408462094-14194-36-git-send-email-jolsa@redhat.com>
Patchwork-id: 88020
O-Subject: [PATCH RHEL7.1 BZ1131394 035/325] tools/perf/build: Add 'autodep' functionality, generate feature test dependencies automatically
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 8b6eb56a9570001634df1d2c7f38e7179a357362
Author: Ingo Molnar <mingo@kernel.org>
Date: Mon Sep 30 13:51:28 2013 +0200

description
===========
Use GCC's -MD feature to generate a dependency file for each feature test .c file,
and include that .d file in the config/feature-checks/Makefile.

This allows us to do two things:

 - speed up feature tests
 - detect removal or changes in build dependencies - including system libraries/headers
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/config/feature-checks/Makefile b/tools/perf/config/feature-checks/Makefile
index b3f6372..4708cca 100644
--- a/tools/perf/config/feature-checks/Makefile
+++ b/tools/perf/config/feature-checks/Makefile
@@ -1,16 +1,20 @@
 
 FILES=test-hello
 
+CC := $(CC) -MD
+
 all: $(FILES)
 
 BUILD = $(CC) -o $(OUTPUT)$@ $@.c
 
 ###############################
 
-test-hello: test-hello.c
+test-hello:
  $(BUILD)
 
+-include *.d */*.d
+
 ###############################
 
 clean:
- rm -f $(FILES)
+ rm -f $(FILES) *.d
-- 
1.7.1